src/RangeParser.php

```diff
@@ -0,0 +1,201 @@
+            } else {
+                throw new InvalidArgumentException('Unsupported range type "' . $type . '"');
+            }
+        }
+    }
+
+    public function parse(?string $value): ?array
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if (!preg_match('/^(?P<open>\[|\()(?P<lower>[^,]*),(?P<upper>[^\)\]]*)(?P<close>\)|\])$/', $value, $matches)) {
+            throw new InvalidArgumentException();
+        }
+
+        $lower = $matches['lower'] ? trim($matches['lower'], '"') : null;
+        $upper = $matches['upper'] ? trim($matches['upper'], '"') : null;
+        $includeLower = $matches['open'] === '[';
+        $includeUpper = $matches['close'] === ']';
+
+        if ($lower === null && $upper === null) {
+            return [null, null];
+        }
+
+        $type = $this->type ?? self::parseType($lower, $upper);
+
+        switch ($type) {
+            case Schema::TYPE_INT_4_RANGE:
+                return self::parseIntRange($lower, $upper, $includeLower, $includeUpper);
+            case Schema::TYPE_INT_8_RANGE:
+                return self::parseBigIntRange($lower, $upper, $includeLower, $includeUpper);
+            case Schema::TYPE_NUM_RANGE:
+                return self::parseNumRange($lower, $upper, $includeLower, $includeUpper);
+            case Schema::TYPE_DATE_RANGE:
+                return self::parseDateRange($lower, $upper, $includeLower, $includeUpper);
+            case Schema::TYPE_TS_RANGE:
+                return self::parseTsRange($lower, $upper, $includeLower, $includeUpper);
+            case Schema::TYPE_TS_TZ_RANGE:
+                return self::parseTsTzRange($lower, $upper, $includeLower, $includeUpper);
+            default:
+                return null;
+        }
+    }
+
+    private static function parseIntRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
+    {
+        $min = $lower === null ? null : (int) $lower;
+        $max = $upper === null ? null : (int) $upper;
+
+        if ($min !== null && $includeLower === false) {
+            $min += 1;
+        }
+
+        if ($max !== null && $includeUpper === false) {
+            $max -= 1;
+        }
+
+        return [$min, $max];
+    }
+
+    private static function parseBigIntRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
+    {
+        if (PHP_INT_SIZE === 8) {
+            return self::parseIntRange($lower, $upper, $includeLower, $includeUpper);
+        }
+
+        [$min, $max] = self::parseNumRange($lower, $upper, $includeLower, $includeUpper);
+
+        if ($min !== null && $includeLower === false) {
+            $min += 1;
+        }
+
+        if ($max !== null && $includeUpper === false) {
+            $max -= 1;
+        }
+
+        return [$min, $max];
+    }
+
+    private static function parseNumRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
```

Comment:
`bool $includeLower, bool $includeUpper` they are not used within the method.

Reply:
I will fix that and add new short immutable methods like `withType`, `asInt`, `asNum` etc